types/schema/data/Proposal.ts

```diff
@@ -167,6 +188,42 @@ export interface LondonProposal extends BaseProposal {
      */
     grey: boolean;
   };
+  /**
+   * @description Proposed energy sources
+   */
+  energy?: {
+    type: Array<'communityOwned' | 'heatPump' | 'solar'>;
+    communityOwned?: {
+      /** @description Proposed total capacity of any on-site community-owned energy generation in megawatts (mW) */
+      capacity: {megawatts: number};
+    };
+    heatPumps?: {
+      /** @description Proposed total capacity of any heat pumps in megawatts (mV) */
+      capacity: {megawatts: number};
+    };
+    solar?: {
+      /** @description Proposed total capacity of any solar energy generation in megawatts (mV) */
+      capacity: {megawatts: number};
+    };
+  };
+  /**
+   * @description Urban Greening Factor Score
+   */
+  urbanGreeningFactor?: {
+    score: number;
```

Comment:
Looks like we can add `@minimum` and `@maximum` annotations here - 

> Scores range from 1 for semi natural
vegetation through to 0 for impermeable sealed surfaces

Source: https://www.london.gov.uk/sites/default/files/ugf_calculator_version_1_march_2021.xlsx & https://www.london.gov.uk/sites/default/files/urban_greening_factor_lpg_pre-consultation_draft.pdf (page 7)

Reply:
Ah this is interesting and doesn't match at all the example values provided (integers 4 & 7) :sweat_smile: 

Will make a note to add `@minimum` & `@maximum` in future, but I think we want to be able to enforce these on the `NumberInput` component first ! Eg here https://editor.planx.dev/opensystemslab/london-data-hub/nodes/8T05wOWOsp/nodes/uBS9tovnxB/edit